From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bjarne Koll <git@lynxplay.dev>
Date: Tue, 20 May 2025 01:41:09 +0200
Subject: [PATCH] Limbo server config

The void server should not be sending *any* packets outside of the login
packet to not confuse the client when quickly switching.

diff --git a/net/minecraft/server/players/PlayerList.java b/net/minecraft/server/players/PlayerList.java
index a7f589970a04e520eb5cbe6dfeb333bc9cd479a8..6473befcc4d8cf5f5bdaf433fab14b3e3a417b84 100644
--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -218,6 +218,7 @@ public abstract class PlayerList {
         component.withStyle(ChatFormatting.YELLOW);
         Component joinMessage = component; // Paper - Adventure
         playerConnection.teleport(player.getX(), player.getY(), player.getZ(), player.getYRot(), player.getXRot());
+        if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.limboServer) return; // KTP - limbo server
         ServerStatus status = this.server.getStatus();
         if (status != null && !cookie.transferred()) {
             player.sendServerStatus(status);
